local_groups/views.py

```diff
@@ -49,18 +55,32 @@ class EventCreateView(LoginRequiredMixin, SuccessMessageMixin, CreateView):
     ''' if settings.EVENT_AUTO_APPROVAL else '''
     Your event was created successfully and is now being reviewed by our team.
     '''
+    permission_required = 'local_groups.add_event'
+
+    def get_local_group(self):
+        local_group = None
+        user = self.request.user
 
-    # Check if user is a group leader and has a valid bsd cons_id
+        if hasattr(user, 'localgroupprofile'):
+            local_group_profile = user.localgroupprofile
+
+            # TODO: support multiple group affiliations?
+            local_group_affiliation = LocalGroupAffiliation.objects.filter(
+                local_group_profile=local_group_profile,
+                local_group__status__exact='approved',
+            ).first()
+            if local_group_affiliation:
+                local_group = local_group_affiliation.local_group
+
+        return local_group
+
+    # Check if user has a valid bsd cons_id
     def can_access(self):
-        is_group_leader = Group.objects.filter(
-            rep_email__iexact=self.request.user.email,
-            status__exact='approved',
-        ).first() is not None
         user = self.request.user
         has_valid_cons_id = hasattr(user, 'bsdprofile') and (
             user.bsdprofile.cons_id != BSDProfile.cons_id_default
         )
-        return is_group_leader and has_valid_cons_id
+        return has_valid_cons_id
```

Comment:
Oh right I forgot about this part, guess you can leave this method in for now.

Reply:
Yeah I was unsure how to handle this so just left it in.